File: 卡码/代码随想录/1/07-回溯/C++/01-组合/LeetCode_0077.cpp
//77. 组合
#include <iostream>
#include <vector>
using namespace std;

class Solution {
	vector<int> path;
	vector<vector<int>> res;
public:
	void backtracking(int n,int k,int index){
		if(k==path.size()){
			res.push_back(path);
			return;
		}
		for(int i=index;i<=n+1-(k-path.size());i++){
			path.push_back(i);
			backtracking(n,k,i+1);
			path.pop_back();
		}
	}
	vector<vector<int>> combine(int n, int k) {
		backtracking(n,k,1);
		return res;
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/07-回溯/Java/backtracking/src/main/java/com/kurumi/backtracking/_01_组合/LeetCode_0077.java
//77. 组合
package com.kurumi.backtracking._01_组合;

import java.util.ArrayList;
import java.util.List;

public class LeetCode_0077 {

}
class Solution {
  List<List<Integer>> res=new ArrayList<>();
  List<Integer> path=new ArrayList<>();
  public void backtracking(int n,int k,int index){
    if(path.size()==k){
      res.add(new ArrayList<>(path));
      return;
    }
    for(int i=index;i<=n+1-(k-path.size());i++){
      path.add(i);
      backtracking(n, k, i+1);
      path.remove(path.size()-1);
    }
  }
  public List<List<Integer>> combine(int n, int k) {
    backtracking(n, k, 1);
    return res;
  }
}
